inc/Engine/CriticalPath/CriticalCSS.php

```diff
@@ -269,51 +277,75 @@ public function get_public_taxonomies() {
 			) AS taxonomies
 			GROUP BY taxonomy"
 		);
-
-		return $rows;
 	}
 
 	/**
 	 * Sets the items for which we generate critical CSS.
 	 *
 	 * @since  2.11
+	 *
+	 * @param string $version Optional. Version of the CPCSS files to generate. Possible values: default, mobile, all.
 	 */
-	public function set_items() {
+	private function set_items( $version = 'default' ) {
 		$page_for_posts = get_option( 'page_for_posts' );
 
 		if ( 'page' === get_option( 'show_on_front' ) && ! empty( $page_for_posts ) ) {
-			$this->items[] = [
-				'type' => 'home',
-				'url'  => get_permalink( get_option( 'page_for_posts' ) ),
+			$this->items['home'] = [
+				'type'  => 'home',
+				'url'   => get_permalink( get_option( 'page_for_posts' ) ),
+				'path'  => 'home.css',
+				'check' => 0,
 			];
 		}
 
 		$post_types = $this->get_public_post_types();
 
 		foreach ( $post_types as $post_type ) {
-			$this->items[] = [
-				'type' => $post_type->post_type,
-				'url'  => get_permalink( $post_type->ID ),
+			$this->items[ $post_type->post_type ] = [
+				'type'  => $post_type->post_type,
+				'url'   => get_permalink( $post_type->ID ),
+				'path'  => "{$post_type->post_type}.css",
+				'check' => 0,
 			];
 		}
 
 		$taxonomies = $this->get_public_taxonomies();
-
 		foreach ( $taxonomies as $taxonomy ) {
-			$this->items[] = [
-				'type' => $taxonomy->taxonomy,
-				'url'  => get_term_link( (int) $taxonomy->ID, $taxonomy->taxonomy ),
+
+			$this->items[ $taxonomy->taxonomy ] = [
+				'type'  => $taxonomy->taxonomy,
+				'url'   => get_term_link( (int) $taxonomy->ID, $taxonomy->taxonomy ),
+				'path'  => "{$taxonomy->taxonomy}.css",
+				'check' => 0,
 			];
 		}
 
+		if ( in_array( $version, [ 'all', 'mobile' ], true ) ) {
+			$mobile_items = [];
+
+			foreach ( $this->items as $key => $value ) {
+				$value['mobile']                 = 1;
+				$value['path']                   = str_replace( '.css', '-mobile.css', $value['path'] );
+				$mobile_items[ "{$key}-mobile" ] = $value;
+			}
+
+			if ( 'mobile' === $version ) {
+				$this->items = $mobile_items;
+			} elseif ( 'all' === $version ) {
+				$this->items = array_merge( $this->items, $mobile_items );
+			}
+		}
+
 		/**
 		 * Filters the array containing the items to send to the critical CSS generator.
 		 *
 		 * @since  2.11.4
 		 *
-		 * @param array $this ->items Array containing the type/url pair for each item to send.
+		 * @param array $items Array containing the type/url pair for each item to send.
 		 */
 		$this->items = (array) apply_filters( 'rocket_cpcss_items', $this->items );
+
+		ksort( $this->items );
```

Comment:
@Tabrisrp `ksort()` is slow. Why do we need this?

Reply:
The goal is to re-order the array to group the 2 versions (default & mobile) for the same page together, instead of doing all the default ones first, then all the mobile ones after

Reply:
Then shouldn't we only do it if we're also doing mobile, i.e. version === 'all'?

Reply:
you're right it's not needed for the other cases. Commit: https://github.com/wp-media/wp-rocket/commit/d04a9865962683ff3a85a390e457fba1a066775c